src/mlpack/core/util/to_lower.hpp

```diff
@@ -0,0 +1,35 @@
+/**
+ * @file to_lower.hpp
+ * @author Himanshu Pathak
+ *
+ * Convert a string to lowercase.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_UTIL_LOWER_STRING_HPP
+#define MLPACK_CORE_UTIL_LOWER_STRING_HPP
+
+namespace mlpack {
+namespace util {
+
+/**
+ * Convert a string to lowercase letters.
+ *
+ * @param str The string to convert.
+ */
+inline std::string ToLower(const std::string& str)
```

Comment:
I wonder if we should pass the output as reference as well, would allow us to do the conversion inplace. Let me know what you think.

Reply:
Sure I will change it.